<launch>

    <arg name="mission_file" default="mission.csv"/>
    <arg name="starting_depth" default="0.0"/>
    <arg name="default_rpm" default="300"/>
    <arg name="goal_tolerance" default="50"/>
    <arg name="latitude" default="59.344427"/>
    <arg name="longitude" default="18.138953"/>
    <arg name="marker_scale" default="20."/>

    <!-- Send a static GPS fix to every new subscriber. Edit latitude and longitude in launch/demo.gps to use your own position. -->
    <node pkg="rostopic" type="rostopic" name="fake_gps_fix"
      args="pub /gps/fix sensor_msgs/NavSatFix --latch
      '{header: {seq: 999, stamp: {secs: 0, nsecs: 0}, frame_id: 'gps'},
        status: {status: 0, service: 1},
        latitude: $(arg latitude),
        longitude: $(arg longitude),
        altitude: 0,
        position_covariance: [3.9561210000000004, 0.0, 0.0, 0.0, 3.9561210000000004, 0.0, 0.0, 0.0, 7.650756],
        position_covariance_type: 2}'" output="screen"/>

    <node pkg="auv_mission_planner" type="mission_planner.py" name="mission_planner" output="screen">
      <param name="mission_file" value="$(arg mission_file)"/>
      <param name="starting_depth" value="$(arg starting_depth)"/>
      <param name="default_rpm" value="$(arg default_rpm)"/>
      <param name="goal_tolerance" value="$(arg goal_tolerance)"/>
      <param name="marker_scale" value="$(arg marker_scale)"/>
    </node>

    <!-- Start rviz with a pre-configured AerialMap instance. It will use the fake GPS fix from above. -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find auv_mission_planner)/rviz/mission_planner.rviz"/>

</launch>
